### PR TITLE
Tag component

### PR DESCRIPTION
**Pull request by NicolasSimmonds**
I added  MarginProps to the Tag interface props, felt like in some cases a dev would like to have them a little bit more separated than they are by default. Let me know if you are good with this.

### CODE REVIEW

File: modules/cactus-web/src/Tag/Tag.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  hidden?: boolean
+  onCloseIconClick?: () => void
+}
+
+const TagBase = React.forwardRef<HTMLSpanElement, TagProps>(
+  ({ id, className, closeOption, children, onCloseIconClick }, ref): React.ReactElement => {
+    return (
+      <span id={id} ref={ref} className={className}>
+        <span className="value-tag__label">{children}</span>
+        {closeOption && <NavigationClose data-role="close" onClick={onCloseIconClick} />}
```
Comment (wilysword) [inline modules/cactus-web/src/Tag/Tag.tsx:23]: I feel like if `onCloseIconClick` is passed this should be displayed even if `closeOption` isn't present.

**Review by wilysword — APPROVED** [review]
Looks good; I have a couple of thoughts, but nothing important enough to withhold approval.

In the storybook, it might be useful to have a way to add tags back without having to leave the page and come back, but I wouldn't say it' important enough to spend a lot of time on it.

File: modules/cactus-web/src/Tag/Tag.tsx (hunk 1 above)
Comment (NicolasSimmonds) [reply modules/cactus-web/src/Tag/Tag.tsx:23]: I like this idea.

Comment (wilysword) [reply modules/cactus-web/src/Tag/Tag.tsx:23]: That's what I would say.

**Review by Dhalton — APPROVED** [review]
Nice work, looks good to me

**Pull request merged**